.github/workflows/pr-review-companion.yml

```diff
@@ -0,0 +1,198 @@
+              exit 1
+          fi
+
+          # Check actor.
+          if [ "$ACTOR" != "$AUTHOR" ]; then
+              ACTOR_PERMISSION=$(gh api https://api.github.com/repos/$REPO/collaborators/$ACTOR/permission --jq .permission)
+
+              if [ "$ACTOR_PERMISSION" != "admin" ]; then
+                  echo "PR actor ($ACTOR) is not an admin, please ping someone for a review."
+                  gh run cancel --repo "$REPO" $RUN_ID
+                  exit 1
+              fi
+          fi
+
+      - name: Checkout (fred)
+        uses: actions/checkout@v4
+        with:
+          path: mdn/fred
+
+      - name: Checkout (content)
+        uses: actions/checkout@v4
+        with:
+          repository: mdn/content
+          path: mdn/content
+
+      - name: Checkout (curriculum)
+        uses: actions/checkout@v4
+        with:
+          repository: mdn/curriculum
+          path: mdn/curriculum
+
+      - name: Checkout (mdn-studio)
+        uses: actions/checkout@v4
+        with:
+          repository: mdn/mdn-studio
+          path: mdn/mdn-studio
+          lfs: true
+          token: ${{ secrets.MDN_STUDIO_PAT }}
+
+      - name: Checkout (generic-content)
+        uses: actions/checkout@v4
+        with:
+          repository: mdn/generic-content
+          path: mdn/generic-content
+
+      - name: Checkout (mdn-contributor-spotlight)
+        uses: actions/checkout@v4
+        with:
+          repository: mdn/mdn-contributor-spotlight
+          path: mdn/mdn-contributor-spotlight
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: "mdn/fred/.nvmrc"
+
+      - name: Build some pages
+        working-directory: mdn/fred
+        env:
+          CONTENT_ROOT: ${{ github.workspace }}/mdn/content/files
+          CONTRIBUTOR_SPOTLIGHT_ROOT: ${{ github.workspace }}/mdn/mdn-contributor-spotlight/contributors
+          BLOG_ROOT: ${{ github.workspace }}/mdn/mdn-studio/content/posts
+          CURRICULUM_ROOT: ${{ github.workspace }}/mdn/curriculum
+          GENERIC_CONTENT_ROOT: ${{ github.workspace }}/mdn/generic-content/files
+
+          # rari
+          LIVE_SAMPLES_BASE_URL: https://live.test.mdnyalp.dev
+          INTERACTIVE_EXAMPLES_BASE_URL: https://interactive-examples.mdn.allizom.net
+          ADDITIONAL_LOCALES_FOR_GENERICS_AND_SPAS: de
+
+          BUILD_LIVE_SAMPLES_BASE_URL: https://live.mdnyalp.dev
+          BUILD_LEGACY_LIVE_SAMPLES_BASE_URL: https://live.mdnyalp.dev
+
+          # Use the stage version of interactive examples.
+          BUILD_INTERACTIVE_EXAMPLES_BASE_URL: https://interactive-examples.mdn.allizom.net
+
+          # Now is not the time to worry about flaws.
+          BUILD_FLAW_LEVELS: "*:ignore"
+
+          # This adds the ability to sign in (stage only for now)
+          REACT_APP_DISABLE_AUTH: true
+
+          # Playground
+          REACT_APP_PLAYGROUND_BASE_HOST: mdnyalp.dev
```

Comment:
@fiji-flo Could you please take a look at these environment variables, as you're more familiar? Do we need these all?